proposals/Imperative-Shadow-DOM-Distribution-API.md

```diff
@@ -1,48 +1,70 @@
 # Imperative Shadow DOM Distribution API
 
-hayato@google.com
+Authors: yuzhehan@chromium.org, hayato@google.com
 
-The straw-man proposal for Imperative Shadow DOM distribution API.
-The context is: [https://github.com/whatwg/html/issues/3534](https://github.com/whatwg/html/issues/3534)
+This is a strawman proposal for Imperative Shadow DOM distribution API.
+For more context, please see [the WhatWG issue](https://github.com/whatwg/html/issues/3534).
 
-# Declarative API and Imperative API should be exclusively used in each shadow tree
+## Motivation
 
-Mixing declarative API and imperative API would be troublesome and can be the cause of confusions for web developers.
-We *can* invent complex rules, however, no one wants to remember complex rules. Also, supporting both in the same shadow tree would make a browser engine complex, which I don't want.
-
-Thus, we don't allow mixing declarative API and imperative API in the same shadow tree.
-Web developers have to show their *opt-in* to use imperative API for each shadow tree.
-
-A shadow root has an associated *slotting*. Web developers can set shadow root's *slotting* to *manual* by specifying it in attachShadow:
-
-```js
-const sr = attachShadow({ mode: ..., (optional) slotting: 'manual' })
+One of the drawbacks of Shadow DOM v1, when compared to Shadow DOM v0, is that web developers have to specify slot= attribute for every shadow host's children (except for elements for the *default* slot).
+ 
+#### Case 1:  Slot attributes is required in markup.
 ```
+<shadow-host>
+  <div slot=slot1></div>
+  <div slot=slot2></div>
+</shadow-host>
+ ```
+Some people would see this as a kind of *ugly* markup.
+Shadow DOM v1 can't explain how `<summary>/<details>` elements can be implemented on the top of the current Web Components technology stack, given that `<details>` element doesn't need slot= attribute.
+Blink has a special logic for some built-in elements to control node-to-slot mapping.
+ 
+#### Case 2: Can’t slot based on condition. 
+``` 
+<shadow-host num-to-show=”2”>
+   <div slot=”slot1”></div>
+   <div slot=”slot1”></div>
+   <div slot=”slot1”></div>
+</shadow-host>
+ ```
+The second issue is that component creators can’t change the slotting behavior based on condition. A component that wants to display a subset of slottable based on a property can’t easily be done via declarative slotting.
```

Comment:
It'd be good to explain the connection to the above markup. Something like "So, there is no way to implement the `num-to-show` attribute in the above example.

It'd be even better if the example were more realistic, e.g. using an attribute like "tab" instead of "num-to-show" and a tag name like "tab-panel" instead of "shadow-host".

Reply:
Thanks for the suggested example. Updated my example to be a better case. Also added an example implementation to showcase the new API.